### DISCUSSION TITLE
Macro issues

### DISCUSSION THREAD

**Discussion by apg1** [KinesisCorporation/Adv360-Pro-ZMK#531] [Q&A]
I'm running into some issues with macros, if I use nickcoutsos' editor, when I try to build, I get 
devicetree error: /keymap/default_layer: undefined node label 'MACRO_MM', 

using the Kinesis editor I get 
devicetree error: /__w/Adv360-Pro-ZMK/Adv360-Pro-ZMK/config/macros.dtsi:5 (column 17): parse error: expected number or parenthesized expression.

What I'm trying to do should be simple, I'm switching the BT input as well as the layer per machine as I have both Windows and OSX and want to have custom layouts for each. Then tying each machine to the center 1-4 keys.

This is an example of one of my macros.
macro_maclaptop: macro_maclaptop{
compatible = "zmk,behavior-macro";
label = "macro_maclaptop";
#binding-cells = <0>;
bindings = <&bt bt_sel 2>, <&tog 4>;
};

Thoughts? When I spoke to Kinesis, I was told it can't do it and it's an advanced keyboard so maybe I should get the standard model. Been using Kinesis since 1997...

**Comment by ReFil** [comment]
Hi, please can you link to your repository so I can see what might be wrong?

**Reply by apg1** [reply]
I blew away my fork and created a new fork last night, tried it a few times, and it still didn't build. 

Tonight I blew away everything in my config, built a clean one with just the new layer, then went back through nick's interface and now, amazingly it built successfully.

I'm assuming I had something bad in my config that wouldn't work, until i cleared it all again (although I did that on my previous fork as well).

The last issue I had was the following:
For Buttons 1 and 4 I used &tog 0 (both windows)
For Buttons 2 and 3 I used &tog 4 (for my OSX layer)

However, when I switch from 2 to 3, it toggles off layer 4. 

I changed the &tog to &to and that seems to resolve the issue. 

So it looks like due to a little magic and me stepping away, it appears to be working.

If you'd still like to see if you can understand whaat was going on I can share my fork and you can look at the failed builds vs the final working one to see if you can ascertain what may have happened.

--apg1
